fulp_modules/bloodsuckers/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,746 @@
+	M.faction |= ROLE_BLOODSUCKER
+
+//This handles the removal of antag huds/special abilities
+/datum/antagonist/bloodsucker/remove_innate_effects(mob/living/mob_override)
+	var/mob/living/M = mob_override || owner.current
+	remove_antag_hud(antag_hud_type, M)
+	M.faction -= ROLE_BLOODSUCKER
+
+///Called by the add_antag_datum() mind proc after the instanced datum is added to the mind's antag_datums list.
+/datum/antagonist/bloodsucker/on_gain()
+	SSticker.mode.bloodsuckers |= owner // Only add after they've been given objectives
+	SSticker.mode.check_start_sunlight() // Start Sunlight if first Vamp
+	SelectFirstName() // Name & Title
+	SelectTitle(am_fledgling = TRUE) 	// If I have a creator, then set as Fledgling.
+	SelectReputation(am_fledgling = TRUE)
+	AssignStarterPowersAndStats() // Give Powers & Stats
+	if(give_objectives)
+		forge_bloodsucker_objectives()
+	update_bloodsucker_icons_added(owner.current, "bloodsucker")
+	. = ..()
+
+///Called by the remove_antag_datum() and remove_all_antag_datums() mind procs for the antag datum to handle its own removal and deletion.
+/datum/antagonist/bloodsucker/on_removal()
+	SSticker.mode.bloodsuckers -= owner
+	SSticker.mode.check_cancel_sunlight() // End Sunlight if last Vamp
+	ClearAllPowersAndStats()// Clear Powers & Stats
+	. = ..()
+
+/datum/antagonist/bloodsucker/greet()
+	var/fullname = ReturnFullName(TRUE)
+	to_chat(owner, "<span class='userdanger'>You are [fullname], a strain of vampire dubbed bloodsucker!</span><br>")
+	owner.announce_objectives()
+	to_chat(owner, "<span class='boldannounce'>* You regenerate your health slowly, you're weak to fire, and you depend on blood to survive. Allow your stolen blood to run too low, and you will find yourself at \
+	risk of being discovered!</span><br>")
+	var/bloodsucker_greet
+	bloodsucker_greet +=  "<span class='boldannounce'>* Other Bloodsuckers are not necessarily your friends, but your survival may depend on cooperation. Betray them at your own discretion and peril.</span><br>"
+	bloodsucker_greet += "<span class='announce'>Bloodsucker Tip: Rest in a <i>Coffin</i> to claim it, and that area, as your lair.</span><br>"
+	bloodsucker_greet += "<span class='announce'>Bloodsucker Tip: Fear the daylight! Solar flares will bombard the station periodically, and only your coffin can guarantee your safety.</span><br>"
+	to_chat(owner, bloodsucker_greet)
+	owner.current.playsound_local(null, 'fulp_modules/bloodsuckers/sounds/BloodsuckerAlert.ogg', 100, FALSE, pressure_affected = FALSE)
+	antag_memory += "Although you were born a mortal, in un-death you earned the name <b>[fullname]</b>.<br>"
+
+/datum/antagonist/bloodsucker/farewell()
+	owner.current.visible_message("[owner.current]'s skin flushes with color, their eyes growing glossier. They look...alive.",\
+			"<span class='userdanger'><FONT size = 3>With a snap, your curse has ended. You are no longer a Bloodsucker. You live once more!</FONT></span>")
```

Comment:
okay, but what if they're an undead race, like a skeleton

Reply:
I'd have to test in game, but can_make_bloodsucker proc should prevent this from happening